src/mame/includes/klax.h

```diff
@@ -58,4 +57,38 @@ class klax_state : public driver_device
 	static const atari_motion_objects_config s_mob_config;
 };
 
+class klax_bootleg_state : public klax_state
+{
+public:
+	klax_bootleg_state(const machine_config &mconfig, device_type type, const char *tag)
+		: klax_state(mconfig, type, tag)
+		, m_audiocpu(*this, "audiocpu")
+		, m_msm(*this, "msm%u", 1)
+		, m_rombank(*this, "rombank")
+		, m_audio_ram(*this, "audioram")
+	{ }
+
+	void klax5bl(machine_config &config);
+
+private:
+	virtual void machine_start() override;
+
+	void m5205_int1(int state);
+
+	void bootleg_sound_map(address_map &map);
+	void klax5bl_map(address_map &map);
+
+	uint16_t audio_ram_r(offs_t offset);
+	void audio_ram_w(offs_t offset, uint16_t data);
+	void audio_sample_w(offs_t offset, uint8_t data);
+	void audio_ctrl_w(uint8_t data);
+
+	required_device<cpu_device> m_audiocpu;
+	required_device_array<msm5205_device, 2> m_msm;
+	required_memory_bank m_rombank;
+	required_shared_ptr<uint8_t> m_audio_ram;
+	uint8_t m_audio_sample[2];
+	bool m_audio_nibble;
```

Comment:
Can you please register these for save states in `machine_start`?  I don’t think the base class actually has members that need saving, so if all the devices used have save state support, the machines could be marked as supporting save states.  The only one I'm not sure about is the Atari Motion Objects, I’m pretty sure the other devices are all save state clean.

Reply:
I'm not sure about `m_last_xpos` in atarimo, but the rest seems ok. There are other games using it that are already marked as having save support anyway (Badlands, Rampart, probably others)